### PR TITLE
Add abort repository migration API call on Github API class

### PR DESCRIPTION
**Pull request by begonaguereca**
**What is changing?** 
Customers have requested a command that could abort a migration that is in progress. 

This PR is adding a method to the `GithubApi` service class that will execute the graph ql query that will abort a migration that is in progress. 

**How is this tested?** 
Unit tests, there is no manual testing of this command yet. 

If you are interested in manually testing the abort migration command  in insomnia you can follow the instructions in the body of [this](https://github.com/github/github/pull/295877) PR. 

**Closes:** https://github.ghe.com/github/octoshift/issues/7784

- [x] Did you write/update appropriate tests
- [ ] Release notes updated (if appropriate)
- [ ] Appropriate logging output
- [ ] Issue linked
- [ ] Docs updated (or issue created)
- [ ] New package licenses are added to `ThirdPartyNotices.txt` (if applicable)

### CODE REVIEW

File: src/Octoshift/Services/GithubApi.cs [hunk 1]
```diff
@@ -1142,4 +1142,37 @@ private static Mannequin BuildMannequin(JToken mannequin)
...
+                    $migrationId: ID!,
+                )";
+        var gql = @"
+                abortRepositoryMigration(
+                    input: { 
+                        migrationId: $migrationId
+                    }
+                ) {
+                    success
+                    }
```
Comment (ArinGhazarian) [inline src/Octoshift/Services/GithubApi.cs:1161]: it looks like this curly brace is extra

Comment (begonaguereca) [reply src/Octoshift/Services/GithubApi.cs:1161]: hahah yes got there eventually! also double quotes around the variable in the JSON blob

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 2]
```diff
@@ -3332,4 +3332,58 @@ public async Task GetEnterpriseServerVersion_Returns_Version()
...
+                )";
+        const string gql = @"
+                abortRepositoryMigration(
+                    input: { 
+                        migrationId: $migrationId
+                    }
+                ) {
+                    sucesss 
+                    }
+                  }";
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3355]: the reason why the test is failing is the payloads need to exactly match, there is a typo in success, and extra space and and extra `}` here. After you fixed those, you can use the `Compact` private method to compare the payload in the client mock setup.

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 3]
```diff
@@ -3332,4 +3332,58 @@ public async Task GetEnterpriseServerVersion_Returns_Version()
...
+                    {{
+                        ""abortRepositoryMigration"": 
+                            {{
+                                ""success"": ""{actualBooleanResponse}""
+                            }}
+                    }}
+            }}");
+
+        _githubClientMock
+            .Setup(m => m.PostGraphQLAsync(url, It.Is<object>(x => x.ToJson() == payload.ToJson()), null))
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3379]:
```suggestion
            .Setup(m => m.PostGraphQLAsync(url, It.Is<object>(x => Compact(x.ToJson()) == Compact(payload.ToJson())), null))
```

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 4]
```diff
@@ -3332,4 +3332,58 @@ public async Task GetEnterpriseServerVersion_Returns_Version()
             .Replace("\\n", "")
             .Replace("\\t", "")
             .Replace(" ", "");
+
+    [Fact]
+    public async Task AbortMigration_Returns_Boolean()
```
Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3337]:
```suggestion
    public async Task AbortMigration_Returns_True_On_Success()
```

Comment (ArinGhazarian) [inline src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3337]: nit: Could you move this test method before the `Compact` method. IMO it's aesthetically more pleasant to have all the `private` methods after all `public` test methods.

Comment (begonaguereca) [reply src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs]: [e226bdd](https://github.com/github/gh-gei/pull/1162/commits/e226bddab3e23c901508ee9b0df81e6f73642196)

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs (hunk 2 above)
Comment (begonaguereca) [reply src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs]: Cool I think i got them all!! Specs are passing now

**Review by ArinGhazarian — APPROVED** [review]
Looks good, just left a nitpick.

**Comment by ArinGhazarian** [comment]
#1165 will fix the flaky CI/build

File: src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs [hunk 5]
```diff
@@ -3332,4 +3383,5 @@ public async Task GetEnterpriseServerVersion_Returns_Version()
             .Replace("\\n", "")
             .Replace("\\t", "")
             .Replace(" ", "");
+
```
[the remark this answers is not in the document]
Comment (begonaguereca) [reply src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3386]: The linting that is enabled is requiring it 🤷🏽‍♀️ , adding it back in

```
/Users/runner/work/gh-gei/gh-gei/src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs(3387,2): error FINALNEWLINE: Fix final newline. Insert '\n'. [/Users/runner/work/gh-gei/gh-gei/src/OctoshiftCLI.Tests/OctoshiftCLI.Tests.csproj]
```

[the remark this answers is not in the document]
Comment (ArinGhazarian) [reply src/OctoshiftCLI.Tests/Octoshift/Services/GithubApiTests.cs:3386]: The linter only requires a new line at the end of a file, the new line I was talking about was added before the last closing brace. But it's no big deal :)

**Pull request merged**